src/DBmysqlIterator.php

```diff
@@ -511,9 +511,6 @@ public function analyseCrit($crit, $bool = "AND")
     {
 
         if (!is_array($crit)) {
-           //if ($_SESSION['glpi_use_mode'] == Session::DEBUG_MODE) {
```

Comment:
Maybe should we activate this depreciation rather than remove the code?

We now have a flag in test to enable strict depreciations, this one could be part of that as far as I remember.

Reply:
I removed this change and tried to apply the deprecation in #17289 .